packages/data/addon/models/navi-fact-response.ts

```diff
@@ -0,0 +1,46 @@
+/**
+ * Copyright 2020, Yahoo Holdings Inc.
+ * Licensed under the terms of the MIT license. See accompanying LICENSE.md file for terms.
+ */
+
+import EmberObject from '@ember/object';
+import { ResponseV1 } from 'navi-data/serializers/facts/interface';
+import moment, { Moment, MomentInput } from 'moment';
+import { TimeDimensionColumn } from './metadata/time-dimension';
+
+function notNull<T>(t: T | null): t is T {
+  return t !== null;
+}
+
+export default class NaviFactResponse extends EmberObject implements ResponseV1 {
+  rows!: ResponseV1['rows'];
+  meta!: ResponseV1['meta'];
+
+  private getTimeDimensionAsMoments(column: TimeDimensionColumn): Moment[] {
+    const { columnMetadata, parameters } = column;
+    const field = columnMetadata.getCanonicalName(parameters);
+    const { rows = [] } = this;
+    return rows
+      .map(row => {
+        const value = row[field];
+        return value ? moment(value as MomentInput) : null;
```

Comment:
```suggestion
        if (value) {
          const dateTime = moment(value as MomentInput);
          return dateTime.isValid() ? dateTime : null;
        }
        return null;
```

Playing around with `moment` I saw that

> ```
> moment.max(moment('abc'), moment()).format() === "Invalid Date"
> ```
 
so we should probably guard against that and add a test for it

Reply:
Yeah, I tested that too. Wasn't sure how likely that was to happen and wanted to keep the checks to a minimum. `null`, `undefined`, `''` are definitely possible.

Reply:
> From version 2.10.5, if an invalid moment is one of the arguments, the result is an invalid moment.